src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,72 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0501: r##"
+This error indicates that a mutable variable which has been captured in a
+closure is being used while it is still captured by the closure. Because the
+closure has borrowed the variable, the variable is not available for use until
```

Comment:
A link would be good, along with a line explaining that borrowing is what's happening in this particular case.